Release/include/pplx/pplx.h

```diff
@@ -156,11 +156,11 @@ class _TaskCollectionImpl
     void _RunAndWait()
     {
         // Limited inlining support for stand-alone tasks
-        if (_M_scheduled_task && _M_pScheduler && _M_pScheduler->unschedule(_M_scheduled_task))
+        auto scheduled_task = _M_scheduled_task.exchange(nullptr);
+        if (scheduled_task && _M_pScheduler && _M_pScheduler->unschedule(scheduled_task))
```

Comment:
Should scheduled task be initialized in the if statement to reduce scope?

Reply:
I did that initially, but it  affects readability - huge long if statement, or if statement over multiple lines.